src/store/userStore.ts

```diff
@@ -0,0 +1,148 @@
+import { action, computed, observable } from 'mobx'
+import { v4 } from 'uuid'
+import AbstractStore from './store'
+
+type UserData = {
+  uuid: string
+  count: number
+  disabled?: { state: boolean; count: number }
+  [name: string]: any
+}
+
+class UserStore extends AbstractStore {
+  constructor() {
+    super()
+    this.init()
+  }
+
+  init = () => {
+    const { usageData } = this
+    if (usageData) {
+      this.localUsageCounter = usageData
+    }
+  }
+
+  @observable localUsageCounter: number = 0
+
+  /**
+   * Returns a string uuid from either local storage or creates and stores a new one.
+   */
+  @computed get uuid(): string {
+    const user = window.localStorage.getItem('uuid')
+    if (!user) {
+      const uuid = v4()
+      window.localStorage.setItem('uuid', uuid)
+      return uuid
+    }
+
+    return user
+  }
+
+  /**
+   * Retruns userData based on {@link UserData} as a string
+   * If userData is not set or there is a uuid mismatch it will return null.
+   */
+  @computed get userData(): UserData | null {
+    const userData = window.localStorage.getItem('userData')
+    if (!userData) {
+      return null
+    }
+    const jsonData = JSON.parse(userData) as UserData
+    const { uuid } = jsonData
+    if (uuid === this.uuid) {
+      return jsonData
+    }
+    window.localStorage.removeItem('userData')
+    return null
+  }
+
+  @computed get usageData(): number | null {
+    const userData = this.userData
+    if (!userData) {
+      return null
+    }
+    return userData.count
+  }
+
+  @computed get isActiveUser(): boolean {
+    const { userData, localUsageCounter } = this
+    if (!userData) return false
+    const { count, disabled } = userData
+    const localCount = Math.max(count, localUsageCounter)
+    if (disabled && disabled.state) {
+      /**
+       * Here we determine when to show the banner to a user who has disabled it.
+       * There are two cases:
+       *
+       * 1. They completed between 2-9 more operations since last disabled
+       * 2. There is random chance they get the banner based on how many operations they completed
+       */
+      return (
+        localCount >= disabled.count + Math.max(Math.random() * 10, 2) ||
+        Math.random() > Math.min(0.9, Math.max(disabled.count * 0.05, 0.5))
+      )
+    } else if (localCount >= 3) {
+      return true
+    }
+    return false
```

Comment:
@gaurangbharti1 What do you make of this as a random way to show users the banner once they have clicked the disable button once.

Reply:
I think it's a good way to not spam the user while giving them a chance to reach us.